catroid/src/main/res/layout/brick_clone.xml

```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Catroid: An on-device visual programming system for Android devices
+  ~ Copyright (C) 2010-2016 The Catrobat Team
+  ~ (<http://developer.catrobat.org/credits>)
+  ~
+  ~ This program is free software: you can redistribute it and/or modify
+  ~ it under the terms of the GNU Affero General Public License as
+  ~ published by the Free Software Foundation, either version 3 of the
+  ~ License, or (at your option) any later version.
+  ~
+  ~ An additional term exception under section 7 of the GNU Affero
+  ~ General Public License, version 3, is available at
+  ~ http://developer.catrobat.org/license_additional_term
+  ~
+  ~ This program is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  ~ GNU Affero General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Affero General Public License
+  ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:gravity="center_vertical"
+    android:orientation="horizontal">
+
+    <CheckBox
+        android:id="@+id/brick_clone_checkbox"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:gravity="center_vertical"
+        android:visibility="gone" />
+
+    <org.catrobat.catroid.ui.BrickLayout
+        android:id="@+id/brick_clone_layout"
+        style="@style/BrickContainer.Control.Medium"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:gravity="center_vertical"
+        android:orientation="horizontal"
+        app:horizontalSpacing="@dimen/brick_flow_layout_horizontal_spacing"
+        app:verticalSpacing="@dimen/brick_flow_layout_vertical_spacing">
+
+        <TextView
+            android:id="@+id/brick_clone_label"
+            style="@style/BrickText.SingleLine"
+            android:text="@string/brick_clone" />
+
+        <Spinner
+            android:id="@+id/brick_clone_spinner"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:clickable="false"
+            android:focusable="false" />
+    </org.catrobat.catroid.ui.BrickLayout>
+
+</LinearLayout>
```

Comment:
The Checkstyle check NewlineAtEndOfFile is currently enabled for java and gradle files only ([see here](
https://github.com/Catrobat/Catroid/blob/98f506ed6e6ad1d19a83d0d6c42744daca1fb309/catroid/config/checkstyle.xml#L208-L211)).

Android Studio is the reason why this check is not enabled for xml files. Creating a new xml file/resource through AS will create a file _not_ ending with a newline. However newly created java files will always end with a newline. I haven't found a setting (which is also shared) yet to automatically add newlines to xml files as well. :crying_cat_face: 

I would prefer to keep the default line endings for created files with AS, but since it's not possible to check **No**NewlineAtEndOfFile with checkstyle afaik, I'm fine to force newlines in xmls as well. It also seems to be the preferred way when working with git.

> Rationale: Any source files and text files in general should end with a line separator to let other easily add new content at the end of file and "diff" command does not show previous lines as changed.